### ISSUE TITLE
Add codelab that shows how to configure an arbitrary base image for a CNB stack

### ISSUE THREAD

**Issue by briandealwis** [GoogleContainerTools/skaffold#4999]
Using an arbitrary image as a base image for buildpacks is finicky.  This issue proposes to create a tutorial demonstrating Skaffold's artifact dependencies to create a consumable base image for a buildpacks image.

----

Buildpacks uses two images when building an application image:
  - A _run image_ serves as the base image for the resulting application image.
  - A _build image_ acts as the host for performing the build.

But run and build images are required to contain some additional metadata to identify the _Stack ID_ and user/group accounts to be used.  A [_stack_](https://buildpacks.io/docs/concepts/components/stack/) is a specification or contract.  For example, the `io.buildpacks.stacks.bionic` stack defines that it provides the same packages as installed on Ubuntu 18.04, and that the build image has the `build-essentials` package.

So the idea is to show how we can annotate an arbitrary image and use it as a run image.  Here we'll use the GCP Ubuntu 18.04 image and annotate it to show that it adheres to the `io.buildpacks.stacks.bionic` stack.
  - create a base image by using a Dockerfile like the following:
```
FROM gcr.io/gcp-runtimes/ubuntu_18_0_4
LABEL io.buildpacks.stack.id io.buildpacks.stacks.bionic
ENV CNB_STACK_ID io.buildpacks.stacks.bionic
ENV CNB_USER_ID=1000
ENV CNB_GROUP_ID=1000
```

and then reference that image in a `skaffold.yaml` that configures the buildpack's [`runImage`](https://skaffold.dev/docs/references/yaml/#build-artifacts-buildpacks-runImage).

cc: @gsquared94

**Comment by gsquared94** [comment]
I tried to create this example app but I immediately ran into 
```
run-image stack id 'io.buildpacks.stacks.bionic' does not match builder stack 'google'
```
and with a little digging [found](https://github.com/GoogleCloudPlatform/buildpacks/blob/8b275117f7c4decdecdd080082f89317dfdb88a0/builders/gcp/base/stack/parent.Dockerfile#L19) maybe the `gcr.io/buildpacks/builder:v1` stackid is `google`. Couldn't find any image on the `gcr.io/buildpacks` registry that supports this stack. 

So I googled and found `cnbs/sample-builder:bionic` that worked with this run image and I was able to build my sample go app. But the deploy failed with:
```
ERROR: failed to launch: determine start command: process type web was not found
```
which I couldn't make much sense of other than it's been reported [before](https://github.com/heroku/java-buildpack/issues/3) around buildpacks.

Also tried `cloudfoundry/cnb:bionic` that matched the stackId but failed with:
```
missing required mixin(s): adduser, apt, ....
```

Was the intention to create **both** builder and run images manually? Otherwise is there a `gcr.io/..` builder image that works with this? Should I install the `mixins` manually in this run image or is `gcr.io/gcp-runtimes/ubuntu_18_0_4` not the right base image to use?
